### PR TITLE
Testing/profile editor

### LINKED ISSUE #53: Test Profile Editor and Profile Components/Actions

### PR DESCRIPTION
**Pull request by rubyr**
## Type of change made:
- [ ] **Bugfix** (non-breaking change which fixes an issue)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [x] **Refactoring** (improves existing functionality)
- [x] **Testing** (tests existing code or improves existing tests)
- [ ] **Styling** (no functionality changed, just visual improvements)

## Description:

This branch adds tests for both the profile editor component, as well as the profile component. It also adjusts the profile component to align with the tests.

## What functionality does this accomplish?

- Tests profile and profileeditor components
- redirects the user to login page if redux store doesn't hold user information

closes #53 

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

### CODE REVIEW

File: code/web/src/modules/user/Profile.js [hunk 1]
```diff
@@ -41,13 +42,15 @@ const Profile = (props) => (
...
     <Grid>
       <GridCell style={{ padding: '2em', textAlign: 'center' }}>
-        <H4 style={{ marginBottom: '0.5em' }}>{props.user.details.name}</H4>
-
-        <p style={{ color: grey2, marginBottom: '0.5em' }}>{props.user.details.email}</p>
-        <p style={{ color: grey2, marginBottom: '3em' }}>{props.user.details.address}</p>
-        <img src={routeImage + props.user.details.image} alt={props.user.details.name} style={{ width: 400 }}/>
-        <p style={{ color: black, marginBottom: '2em', marginTop: '2em' }}>{props.user.details.bio}</p>
-
+        {props.user.details ?
```
Comment (J-Poulter) [inline code/web/src/modules/user/Profile.js:45]: What situations was this conditional added for?

Comment (rubyr) [reply code/web/src/modules/user/Profile.js:45]: Sometimes the user data would get dropped from the store, this redirects the user to login if that ever happens

Comment (SMJ289) [reply code/web/src/modules/user/Profile.js:45]: That is a good fallback.

Comment (SMJ289) [reply code/web/src/modules/user/Profile.js:45]: I think that is a good fallback to have

**Pull request merged**